api/client.go

```diff
@@ -507,23 +520,29 @@ func (c *Client) getSinkPathFromAgent(agentAddress string) (string, error) {
 }
 
 // starts a go routine to poll the specified file for a token
-func (c *Client) pollFileForToken(filePath string) {
+func (c *Client) pollFileForToken(filePath string, quit chan int) {
```

Comment:
I would either remove the Client receiver (i.e. make it a standalone function), or (preferably) remove the arguments, since every time we invoke it we pass the same args and they're available within Client itself.